user-service/src/main/java/rs/edu/raf/banka1/services/UserServiceImpl.java

```diff
@@ -206,4 +202,25 @@ private List<PermissionDto> extractPermissionsFromUser(User user) {
         return user.getPermissions().stream().
                 map(permissionMapper::permissionToPermissionDto).collect(Collectors.toList());
     }
+
+    @Override
+    public Boolean sendResetPasswordEmail(String email) {
+        Optional<User> optionalUser = userRepository.findByEmail(email);
+        if (optionalUser.isEmpty()) return false;
+        User user = optionalUser.get();
+        String resetPasswordToken = UUID.randomUUID().toString();
+        user.setResetPasswordToken(resetPasswordToken);
+        userRepository.save(user);
+        return emailService.sendEmail(email, "RAF Banka - Password reset",
+                "Visit this URL to reset your password: http://localhost:" + frontPort + "/user/reset-password/" + resetPasswordToken);
+    }
+
+    @Override
+    public NewPasswordResponse setNewPassword(String token, String password) throws NoSuchElementException {
```

Comment:
Why throws NoSuchElementException? Did you check the api call returns 400 on missing token, like it says on your swagger documentation? This exception will delegate to GlobalExceptionHandler, where I see a 404 is returned with this exception.